### PR TITLE
feat:655 pubsublitetobigtable java template

### PR DESCRIPTION
**Pull request by balajiss2**
feat:655 pubsublitetobigtable java template
#655

### CODE REVIEW

File: java/.ci/Jenkinsfile [hunk 1]
```diff
@@ -769,7 +769,28 @@ pipeline {
...
             }
-            
+            stage('Parallel Execution 10'){
+                                        parallel{
+                                            stage('PubSubLite To BigTable') {
+                                                steps{
+                                                    sh '''
+                                                        export GCP_PROJECT=yadavaja-sandbox
+                                                        export REGION=us-west1
+                                                        export GCS_STAGING_LOCATION="gs://python-dataproc-templates-temp"
```
Comment (vanshaj-bhatia) [inline java/.ci/Jenkinsfile:779]: Please remove these, as they are already present in the Jenkins Global Variables.

File: java/.ci/Jenkinsfile [hunk 2]
```diff
@@ -769,7 +769,28 @@ pipeline {
...
+                                                        cd java
+                                                        bin/start.sh \
+                                                        -- --template PUBSUBLITETOBIGTABLE \
+                                                        --templateProperty pubsub.input.project.id=yadavaja-sandbox \
```
Comment (vanshaj-bhatia) [inline java/.ci/Jenkinsfile:783]: Avoid using the project name. Please use $GCP_PROJECT instead.

File: java/.ci/Jenkinsfile [hunk 3]
```diff
@@ -769,7 +769,28 @@ pipeline {
...
+                                                        --templateProperty pubsub.input.subscription=projects/617357862702/locations/us-west1/subscriptions/pubsublitetobigtable-test-subscription2 \
```
Comment (vanshaj-bhatia) [inline java/.ci/Jenkinsfile:784]: Substitute, 617357862702 with $GCP_PROJECT_NUMBER

File: java/.ci/Jenkinsfile [hunk 4]
```diff
@@ -769,7 +769,28 @@ pipeline {
...
+                                                        --templateProperty pubsublite.checkpoint.location=dataproc-template-test1/checkpointlocation \
+                                                        --templateProperty pubsub.bigtable.output.project.id=yadavaja-sandbox \
```
Comment (vanshaj-bhatia) [inline java/.ci/Jenkinsfile:786]: Avoid using the project name. Please use $GCP_PROJECT instead.

File: java/.ci/Jenkinsfile [hunk 5]
```diff
@@ -769,7 +769,28 @@ pipeline {
...
+                                                        --templateProperty pubsub.bigtable.output.instance.id=bt-templates-test \
```
Comment (vanshaj-bhatia) [inline java/.ci/Jenkinsfile:787]: Substitute bt-templates-test with $ENV_TEST_BIGTABLE_INSTANCE

File: java/.ci/Jenkinsfile
Comment (vanshaj-bhatia) [inline java/.ci/Jenkinsfile:1]: Add export SKIP_BUILD=true

File: java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToBigTable.java [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+  @Override
+  public void runTemplate() throws InterruptedException, TimeoutException, StreamingQueryException {
+
+    validateInput();
+    // Initialize the Spark session
+    SparkSession spark =
+        SparkSession.builder().appName("Spark PubSubLiteToGCS Demo Job").getOrCreate();
+
+    // Stream data from Kafka topic
+    Dataset<Row> df =
```
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToBigTable.java:76]: Kafka topic?

File: java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToBigTable.java [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+    pubsubInputSubscription = getProperties().getProperty(PUBSUB_INPUT_SUBSCRIPTION_PROP);
+    timeoutMs = Long.parseLong(getProperties().getProperty(PUBSUB_TIMEOUT_MS_PROP));
+    streamingDuration =
+        Integer.parseInt(getProperties().getProperty(PUBSUB_STREAMING_DURATION_SECONDS_PROP));
+    pubsubCheckpointLocation = getProperties().getProperty(PUBSUBLITE_CHECKPOINT_LOCATION_PROP);
+    pubSubBigTableOutputInstanceId =
+        getProperties().getProperty(PUBSUB_BIGTABLE_OUTPUT_INSTANCE_ID_PROP);
+    pubSubBigTableOutputProjectId =
+        getProperties().getProperty(PUBSUB_BIGTABLE_OUTPUT_PROJECT_ID_PROP);
+    pubSubBigTableOutputTable = getProperties().getProperty(PUBSUB_BIGTABLE_OUTPUT_TABLE_PROP);
```
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToBigTable.java:63]: Any specific reason, why are we reusing PUBSUB_* template properties?

File: java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/README.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+GCP_PROJECT=<gcp-project-id> \
+REGION=<region> \
+SUBNET=<subnet> \
+GCS_STAGING_LOCATION=<gcs-staging-bucket-folder>
+# ID of Dataproc cluster running permanent history server to access historic logs.
+#export HISTORY_SERVER_CLUSTER=<gcp-project-dataproc-history-server-id>
+
+
+bin/start.sh \
+-- --template PUBSUBTOBIGTABLE \
```
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/README.md:15]: Update to PUBSUBLITETOBIGTABLE

**Review by vanshaj-bhatia — CHANGES_REQUESTED** [review]

File: java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToBigTable.java (hunk 7 above)
Comment (balajiss2) [reply java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToBigTable.java:63]: All the properties are applicable to both Pubsub and Pubsublite template except for it has one checkpoint variable in addition. With the intent for reducing redundant variables, i have reused. Moreover the names are also almost same except for pubsub and pubsublite. If there is any other name, i would have definitely used new variables. But if the standard is to create new variables, i can go ahead and create them. Pls let me know

File: java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/README.md (hunk 8 above)
Comment (balajiss2) [reply java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/README.md]: Changed and committed

File: java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToBigTable.java (hunk 6 above)
Comment (balajiss2) [reply java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToBigTable.java]: changed to pubsub and committed

File: java/.ci/Jenkinsfile
Comment (balajiss2) [reply java/.ci/Jenkinsfile:1]: Added and committed

File: java/.ci/Jenkinsfile (hunk 4 above)
Comment (balajiss2) [reply java/.ci/Jenkinsfile]: Changed and committed

File: java/.ci/Jenkinsfile (hunk 3 above)
Comment (balajiss2) [reply java/.ci/Jenkinsfile]: Changed and committed

File: java/.ci/Jenkinsfile (hunk 1 above)
Comment (balajiss2) [reply java/.ci/Jenkinsfile]: Removed and committed

File: java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToBigTable.java (hunk 7 above)
Comment (vanshaj-bhatia) [reply java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToBigTable.java:63]: I believe, it is misleading.  e.g. pubsubInputSubscription (PUBSUB_INPUT_SUBSCRIPTION_PROP), it should be clear that it is not related to pubsub but pubsub lite.

Comment (vanshaj-bhatia) [reply java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToBigTable.java:63]: I would suggest renaming to convey the correct usage.

Comment (balajiss2) [reply java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToBigTable.java]: Made the requested changes and commited

**Review by vanshaj-bhatia — APPROVED** [review]
LGTM!

**Comment by vanshaj-bhatia** [comment]
Good job @balajiss2!

**Pull request merged**